packages/wallet/backend/src/account/service.ts

```diff
@@ -70,18 +71,27 @@ export class AccountService implements IAccountService {
       throw new NotFound()
     }
 
-    const result = await this.gateHubClient.createWallet(
-      user.gateHubUserId,
-      args.name
-    )
+    let gateHubWalletId
+    if (args.isDefaultCardsAccount) {
+      const result = await this.gateHubClient.getWalletForUser(
+        user.gateHubUserId
+      )
+      gateHubWalletId = result.wallets[0].address
```

Comment:
Just for my clarity: With the `isDefaultCardsAccount`: if the user already has a managed user created, we fetch the wallet, otherwise the create one?

Reply:
If the user already has a managed user created **and** the call to create account has `isDefaultCardsAccount` set to `true`, then we would use the already existing GateHub wallet of the managed user instead of creating a new one. Otherwise, if `isDefaultCardsAccount` is false (which happens in all cases unless specified true), then it means that some user just wants to create an account that is not related to any card so we would just create a new GateHub wallet as well.